fdbserver/IPager.h

```diff
@@ -76,16 +77,69 @@ static const std::vector<std::pair<PagerEvents, PagerEventReasons>> L0PossibleEv
 	{ PagerEvents::PageWrite, PagerEventReasons::MetaData },
 };
 
-// Represents a block of memory in a 4096-byte aligned location held by an Arena.
+enum EncodingType : uint8_t {
+	XXHash64 = 0,
+	// For testing purposes
+	XOREncryption = 1
+};
+
+enum PageType : uint8_t {
+	HeaderPage = 0,
+	BackupHeaderPage = 1,
+	BTreeNode = 2,
+	BTreeSuperNode = 3,
+	QueuePageStandalone = 4,
+	QueuePageInExtent = 5
+};
+
+// Encryption key ID
+typedef uint64_t KeyID;
+
+// Encryption key and secret string
+struct KeyIDWithSecret {
+	KeyID id;
+	Key secret;
+};
+
+// Interface used by pager to get encryption keys by ID when reading pages from disk
+// and by the BTree to get encryption keys to use for new pages
+class IEncryptionKeyProvider {
+public:
+	virtual Future<Key> getSecretByKeyID(KeyID keyID) = 0;
+	virtual Future<KeyIDWithSecret> selectKey(const KeyRef begin, const KeyRef end) = 0;
+};
+
+// ArenaPage represents a data page meant to be stored on disk, located in a block of
+// 4k-aligned memory held by an Arena
+//
+// Page Format:
+//    VersionHeader - describes main header version, encoding type, and offsets of subheaders and payload.
```

Comment:
nit: possible to add field width (bits) for better clarification?

Reply:
In this comment?  The sizes are variable.